Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm

```diff
@@ -902,6 +911,25 @@ - (void)documentPickerWasCancelled:(UIDocumentPickerViewController *)documentPic
     [self _cancel];
 }
 
+#pragma mark - PHPickerViewControllerDelegate implementation
+
+- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results
+{
+    [self _processPickerResults:results successBlock:^(NSArray<_WKFileUploadItem *> *items) {
+        ensureOnMainRunLoop([self, strongSelf = retainPtr(self), items = retainPtr(items)] {
+            [strongSelf->_view _removeTemporaryDirectoriesWhenDeallocated:WTFMove(strongSelf->_temporaryUploadedFileURLs)];
+            strongSelf->_temporaryUploadedFileURLs = { };
```

Comment:
Nit - you can skip this if you've already moved `_temporaryUploadedFileURLs ` above. I would use `std::exchange(strongSelf->_temporaryUploadedFileURLs, { })` when calling `-_removeTemporaryDirectoriesWhenDeallocated:` as well, to make this more explicit.

Reply:
skip what exactly?

Reply:
do you suggest
```
[strongSelf->_view _removeTemporaryDirectoriesWhenDeallocated:WTFMove(std::exchange(strongSelf->_temporaryUploadedFileURLs, { }))];
```

instead of
```
[strongSelf->_view _removeTemporaryDirectoriesWhenDeallocated:WTFMove(strongSelf->_temporaryUploadedFileURLs)];
            strongSelf->_temporaryUploadedFileURLs = { };
```

?

Reply:
My suggestion is:
```
[strongSelf->_view _removeTemporaryDirectoriesWhenDeallocated:std::exchange(strongSelf->_temporaryUploadedFileURLs, { })];
```